server/src/extent_tree.c

```diff
@@ -317,18 +317,21 @@ int extent_tree_truncate(
     struct extent_tree* tree, /* tree to truncate */
     unsigned long size)       /* size to truncate extents to */
 {
+    unsigned long target_offset = size - 1;
+
     /* lock the tree for reading */
     extent_tree_wrlock(tree);
 
     /* iterate from node with maximum extent backwards until
      * we find an extent below the truncated size */
     struct extent_tree_node* node = RB_MAX(inttree, &tree->head);
-    while (node != NULL && node->end >= size) {
```

Comment:
Actually, I think this check is still ok as is, for example truncate(size=10) can keep any extent whose end is 9 or less.  However, any extent whose end=10 or more should be updated or removed.